lib/fluent/plugin/out_cloudwatch_logs.rb

```diff
@@ -255,7 +255,7 @@ def create_log_stream(group_name, stream_name)
     def log_group_exists?(group_name)
       if @sequence_tokens[group_name]
         true
-      elsif @logs.describe_log_groups.log_groups.any? {|i| i.log_group_name == group_name }
+      elsif @logs.describe_log_groups({log_group_name_prefix: group_name}).log_groups != []
```

Comment:
This ensures only that a group starting with `group_name` exists. How about the following?

```ruby
elsif @logs.describe_log_groups({log_group_name_prefix: group_name}).log_groups.any? {|i| i.log_group_name == group_name }
```